shared/eip712/eip712.go

```diff
@@ -0,0 +1,95 @@
+package eip712
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+type EIP712Components struct {
+	R [32]byte `json:"r"`
+	S [32]byte `json:"s"`
+	V uint8    `json:"v"`
+}
+
+const EIP712Length = 65
+
+// Pretty print for EIP712Components
+func (e *EIP712Components) Print() {
+	fmt.Println("EIP712 Components:")
+	fmt.Printf("R: %x\n", e.R)
+	fmt.Printf("S: %x\n", e.S)
+	fmt.Printf("V: %d\n", e.V)
+}
+
+// String returns a hexadecimal string representation of EIP712Components
+func (e *EIP712Components) String() string {
+	out, err := e.MarshalText()
+	if err != nil {
+		// MarshalText should never return an error
+		panic(err)
+	}
+	return fmt.Sprintf("%x", out)
+
+}
+
+// UnmarshalText verifies the length of a decoded EIP-712 signature and assigns the appropriate bytes to R/S/V
+func (e *EIP712Components) UnmarshalText(inp []byte) error {
```

Comment:
ok this is gonna be moderately confusing, but we should expect `inp` to be a 0x prefixed array of bytes with a length of exactly 130....

UnmarshalText expects "UTF-8-encoded text". That means a string of the format `0x[0-9a-fA-F]{130}`

The `encoding` package also has an `UnmarshalBinary` and `MarshalBinary` interface which is what you actually implemented here.

Reply:
(and MarshalText should do the reverse, of course)